src/components/Form/WysiwygControls/HeaderSelect.js

```diff
@@ -1,16 +1,28 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-function HeaderSelect({ value, handleSelectChange, headerBlockTypes }) {
+function HeaderSelect({ editor, handleSelectChange, headerBlockTypes }) {
+  let headerValue = JSON.stringify('') // default to normal text
+
+  const headerLevels = headerBlockTypes.map(
+    (heading) => JSON.parse(heading.value)?.level,
+  )
+  const matchedLevel = headerLevels.find((lvl) =>
+    editor?.isActive('heading', { level: lvl }),
+  )
+  if (matchedLevel) {
+    headerValue = JSON.stringify({ level: matchedLevel })
+  }
+
   const onToggle = (event) => {
     event.preventDefault()
-    const style = event.target.value
-    handleSelectChange(event, style)
+    const parsedValue = JSON.parse(event.target.value)
```

Comment:
One suggestion: given how tip TipTap operates you can probably avoid the parsing/stringifying here if we switch over to a drop down menu (rather than a native select) and use buttons with `() => editor.chain().focus().toggleHeading({ level: 1 }).run()`
Let me know if you want to chat about this 👍

Reply:
Per our discussion I have created this ticket: https://firstvoices.atlassian.net/browse/FW-6665 to convert to a more tiptap-like structure after the draftJS code has been removed